R/parameters.R

```diff
@@ -0,0 +1,196 @@
+  if (!is.list(overrides) && !is.null(overrides)) {
+    stop("overrides must be a list")
+  }
+
+  for (name in names(overrides)) {
+    if (!(name %in% names(pars))) {
+      stop(paste("unknown parameter", name, sep = " " ))
+    }
+    pars[[name]] <- overrides[[name]]
+  }
+
+  if (pars$beta < 0) {
+    stop("'beta' must be positive")
+  }
+
+  if (pars$nu < 0) {
+    stop("'nu' must be positive")
+  }
+
+  if (pars$mu < 0) {
+    stop("'mu' must be positive")
+  }
+
+  if (pars$prop_immune > 0 || pars$prop_immune < 0) {
+    stop("'prop_immune' must be between 0 and 1 (inclusive)")
+  }
+
+  if (pars$N <= 0) {
+    stop("'N' must be positive")
+  }
+
+  if (pars$num <= 0) {
+    stop("'num' must be positive")
+  }
+
+  if (pars$I0 > pars$N || pars$I0 < 0) {
+    stop("'I0' must be positive and never greater than N")
+  }
+
+  if (pars$dt <= 0) {
+    stop("'dt' must be positive and greater than 0")
+  }
+
+  if (pars$n_deaths_S < 0) {
+    stop("'n_deaths_S' must be positive and greater than or equal to 0")
+  }
+
+  if (pars$n_infections_S < 0) {
+    stop("'n_infections_S' must be positive and greater than or equal to 0")
+  }
+
+  if (pars$n_deaths_I < 0) {
+    stop("'n_deaths_I' must be positive and greater than or equal to 0")
+  }
+
+  if (pars$n_recoveries_I < 0) {
+    stop("'n_recoveries_I' must be positive and greater than or equal to 0")
+  }
+
+  if (pars$n_deaths_R < 0) {
+    stop("'n_deaths_R' must be positive and greater than or equal to 0")
+  }
+
+  if (pars$n_births < 0) {
+    stop("'n_births' must be positive and greater than or equal to 0")
+  }
+
+  if (pars$average_age <= 0) {
+    stop("'average_age' must be positive and greater than 0")
+  }
+
+  if (pars$infection_rate < 0) {
+    stop("'infection_rate' must be positive or equal to 0")
+  }
+
+  if (pars$severe_infection_rate < 0) {
+    stop("'severe_infection_rate' must be positive or equal to 0")
+  }
+
+  if (pars$recovery_rate < 0) {
+    stop("'recovery_rate' must be positive or equal to 0")
+  }
+
+  if (pars$age_rate < 0) {
+    stop("'age_rate' must be positive or equal to 0")
+  }
+
+  if (pars$location_rate < 0) {
+    stop("'location_rate' must be positive or equal to 0")
+  }
+
+  pars
+
+}
+
+sir_model_parameters_defaults <- function() {
+
+  pars <- list(
+    # Compartmental
+    beta = 0.5,
+    nu = 0.3,
+    mu = 0.001,
+    prop_immune = 0,
+    I0_at_steady_state = FALSE,
+    N = 10000,
+    num = 100,
+    I0 = 5,
+    dt = 0.01,
+    n_deaths_S = 0,
+    n_infections_S = 0,
+    n_deaths_I = 0,
+    n_recoveries_I = 0,
+    n_deaths_R = 0,
+    n_births = 0,
+    # individual only
+    novariations = TRUE,
+    average_age = 20,
+    includeage = FALSE,
+    includebirth = FALSE,
+    includeimmune = FALSE,
+    includelocation = FALSE,
+    infection_rate = 0.4,
+    severe_infection_rate = 0.1,
+    recovery_rate = 0.8,
+    age_rate = 0.2,
+    location_rate = 0.2)
+
+  pars
+}
+
+default_probs <- function() {
```

Comment:
Can be grabbed from squire to maintain parameter consistency `squire::default_probs()`